### ISSUE TITLE
the cursor problem in zsh terminal

### ISSUE THREAD

**Issue by knuff22** [microsoft/vscode#158241]
Type: <b>Bug</b>

1. Update VScode to 1.70.0 or 1.70.1.
2. Open Terminal in zsh.

The cursor comes to annoying location in the prompt.

With this enviroment variable set,  
```
echo $PS1
%n@%m %1~ %#
```
The cursor comes in between my machine name and the current directory.

VS Code version: Code 1.70.1 (6d9b74a70ca9c7733b29f0456fd8195364076dda, 2022-08-10T06:08:46.959Z)
OS version: Darwin arm64 21.6.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|5, 4, 3|
|Memory (System)|16.00GB (0.28GB free)|
|Process Argv|--crash-reporter-id fa0e43fd-e407-48e1-863f-9ca0d23ebb50|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (11)</summary>

Extension|Author (truncated)|Version
---|---|---
gitlens|eam|12.1.2
shell-format|fox|7.2.2
hcl|has|0.2.1
terraform-preview|has|2022.8.1114
git-graph|mhu|1.30.0
vsliveshare|ms-|1.0.5683
copy-json-path|nid|0.2.0
vscode-yaml|red|1.9.1
partial-diff|ryu|1.4.3
shellcheck|tim|0.20.0
markdown-all-in-one|yzh|3.4.3

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
wslgetstarted:30449410
cppdebug:30492333
vscaac:30438847
pylanb8912:30545647
vsclangdf:30486550
c4g48928:30535728
hb751961:30543456
dsvsc012cf:30540253
azure-dev_surveyonecf:30545436

```

</details>

**Comment by gjsjohnmurray** [comment]
Do you still have the problem after setting `"terminal.integrated.shellIntegration.enabled": false`?

**Comment by knuff22** [comment]
> Do you still have the problem after setting `"terminal.integrated.shellIntegration.enabled": false`? 

Thanks! It worked!

Does it mean I can't use this feature in my environment?

**Comment by meganrogge** [comment]
what's the value you have set for your `PS1` in your rc files?

**Comment by meganrogge** [comment]
probably same cause as #157912

**Comment by knuff22** [comment]
> what's the value for `PS1` if you echo in an external terminal?

Echoed in my iTerm2 and it is same as written above:
`%n@%m %1~ %#`

And FYI I didn't specify the value for PS1 in my .zshrc file, i.e. that is default.

**Comment by knuff22** [comment]
@meganrogge 
It seems to be fixed in my current environment. 

Version: 1.71.0
Commit: 784b0177c56c607789f9638da7b6bf3230d47a8c
Date: 2022-09-01T07:25:25.516Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin arm64 21.6.0
Sandboxed: No

**Comment by tpak** [comment]
@meganrogge this is still happening for me when I set 
"terminal.integrated.shellIntegration.enabled": true

```
Version: 1.71.1
Commit: e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb
Date: 2022-09-08T19:41:58.611Z (6 days ago)
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin arm64 21.6.0
Sandboxed: No
```

I get this gibberish when I open the terminal:
```
/Users/chris/.zshrc:source:107: no such file or directory: /var/folders/kf/gtth807d25b46c0cgxlm5j3c0000gn/T/vscode-zsh/.zim/init.zsh
/Users/chris/.zlogin:source:7: no such file or directory: /var/folders/kf/gtth807d25b46c0cgxlm5j3c0000gn/T/vscode-zsh/.zim/login_init.zsh
chris@chriss-mbp-16 infrastructure % 
```

Is this the same bug as here: https://github.com/microsoft/vscode/issues/157912
I did add a comment there a while back. But I am not sure this should be closed unless this is a duplicate of #157912
